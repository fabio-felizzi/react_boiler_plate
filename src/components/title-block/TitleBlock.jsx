import React from 'react';
import PropTypes from 'prop-types';

const TitleBlock = ({
  title,
  subtitle,
  children,
  backgroundColor,
  className,
}) => (
  <div className={`title-block ${backgroundColor} ${className}`}>
    <div className="content">
      <h1 className="title">
        {title}
      </h1>
      {subtitle && (
        <h2 className="subtitle">
          {subtitle}
        </h2>
      )}
      {children}
    </div>
  </div>
);

TitleBlock.defaultProps = {
  subtitle: '',
  backgroundColor: 'white',
  children: null,
  className: '',
};

TitleBlock.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  backgroundColor: PropTypes.oneOf(['black', 'white']),
  children: PropTypes.element,
  className: PropTypes.string,
};

export default TitleBlock;
