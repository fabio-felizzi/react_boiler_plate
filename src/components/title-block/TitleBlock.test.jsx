import React from 'react';
import { shallow } from 'enzyme';
import TitleBlock from './TitleBlock';

const componentProps = {
  title: 'Component title',
  subtitle: 'Component subtitle',
};

// eslint-disable-next-line react/jsx-props-no-spreading
const componentShallow = shallow(<TitleBlock {...componentProps} />);

describe('TitleBlock component', () => {
  describe('Component Shapshot', () => {
    test('should match stored snapshot', () => {
      expect(componentShallow).toMatchSnapshot();
    });
  });

  describe('Component Render', () => {
    test('should render the title', () => {
      expect(componentShallow.find('h1')).toHaveLength(1);
    });

    test('should render the subtitle', () => {
      expect(componentShallow.find('h2')).toHaveLength(1);
    });
  });
});
