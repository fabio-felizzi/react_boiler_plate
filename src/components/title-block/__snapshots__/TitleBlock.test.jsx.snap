// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBlock component Component Shapshot should match stored snapshot 1`] = `
<div
  className="title-block white undefined"
>
  <div
    className="content"
  >
    <h1
      className="title"
    >
      Component title
    </h1>
    <h2
      className="subtitle"
    >
      Component subtitle
    </h2>
  </div>
</div>
`;
